src/sentry/search/events/fields.py

```diff
@@ -3065,7 +3115,16 @@ def resolve_function(
 
         for arg in snql_function.args:
             if isinstance(arg, ColumnArg):
-                arguments[arg.name] = self.resolve_column(arguments[arg.name])
+                if (
+                    arguments[arg.name] in NumericColumn.numeric_array_columns
+                    and isinstance(arg, NumericColumn)
+                    and not isinstance(combinator, SnQLArrayCombinator)
+                ):
+                    arguments[arg.name] = Function(
+                        "arrayJoin", [self.resolve_column(arguments[arg.name])]
+                    )
+                else:
```

Comment:
This isn't great, I wonder if we can switch to using `histogramArray` in some cases to leverage the combinator.

Reply:
Ooh that's a good idea, going to backlog something for that since it'd require changing how `get_histogram_column` works which is still shared between snql + json